pkg/syncer/fullsync.go

```diff
@@ -194,8 +202,12 @@ func fullSyncDeleteVolumes(ctx context.Context, volumeIDDeleteArray []cnstypes.C
 	// Create map for easy lookup
 	for _, pv := range currentK8sPV {
 		var volumeHandle string
-		var err error
-		if migrationFeatureStateForFullSync && pv.Spec.VsphereVolume != nil {
+		if pv.Spec.VsphereVolume != nil && migrationFeatureStateForFullSync {
+			volumeMigrationService, err := migration.GetVolumeMigrationService(ctx, &metadataSyncer.volumeManager, metadataSyncer.configInfo.Cfg)
```

Comment:
move out of loop.

Reply:
Removed the instantiation here, since the VMS is already initialized at the beginning of full sync